runtime/src/snapshot_utils.rs

```diff
@@ -1723,14 +2234,44 @@ fn verify_unpacked_snapshots_dir_and_version(
     );
 
     let snapshot_version = unpacked_snapshots_dir_and_version.snapshot_version;
-    let mut bank_snapshots =
-        get_bank_snapshots_post(&unpacked_snapshots_dir_and_version.unpacked_snapshots_dir);
-    if bank_snapshots.len() > 1 {
-        return Err(get_io_error("invalid snapshot format"));
-    }
-    let root_paths = bank_snapshots
-        .pop()
-        .ok_or_else(|| get_io_error("No snapshots found in snapshots directory"))?;
+    let root_paths = match unpacked_snapshots_dir_and_version.snapshot_from {
+        SnapshotFrom::Archive => {
+            let mut bank_snapshots =
+                get_bank_snapshots_post(&unpacked_snapshots_dir_and_version.unpacked_snapshots_dir);
+            if bank_snapshots.len() > 1 {
+                return Err(get_io_error("invalid snapshot format"));
+            }
+            bank_snapshots
+                .pop()
+                .ok_or_else(|| get_io_error("No snapshots found in snapshots directory"))?
+        }
+        SnapshotFrom::File => {
+            let snapshot_file_path = &unpacked_snapshots_dir_and_version.snapshot_file_path;
+            let (slot, _) =
+                parse_snapshot_filename(snapshot_file_path.file_name().unwrap().to_str().unwrap())
+                    .ok_or_else(|| {
+                        info!("Invalid snapshot filename {}", snapshot_file_path.display());
+                        SnapshotError::InvalidAccountAppendvecFile
+                    })
+                    .unwrap();
+            BankSnapshotInfo {
+                slot,
+                snapshot_path: unpacked_snapshots_dir_and_version
+                    .snapshot_file_path
+                    .clone(),
+                snapshot_type: BankSnapshotType::Pre,
+                status_cache_path: snapshot_file_path
+                    .parent()
+                    .unwrap()
+                    .join(SNAPSHOT_STATUS_CACHE_FILENAME),
+            }
+        }
+        SnapshotFrom::None => {
```

Comment:
Only place this is used, as far as I can tell. Better to just have snapshot_from be an option like my earlier comment suggested.